### PR TITLE
Do fail the run if no tomli and there is tool.versioneer section in the pyproject.toml

### PR DESCRIPTION
**Pull request by yarikoptic**
<details>
<summary>
Because now what could happen is that it would still fall through to try to load from setup.cfg and just fail with non-informative:
</summary> 

	Running "build_script" scripts
	python -m pip install -q -r requirements-devel.txt
	  ERROR: Command errored out with exit status 1:
	   command: /home/appveyor/dlvenv/bin/python /home/appveyor/DLTMP/tmpjm4hxin6 get_requires_for_build_wheel /home/appveyor/DLTMP/tmpbz0ku69x
		   cwd: /home/appveyor/projects/datalad-container
	  Complete output (27 lines):
	  Traceback (most recent call last):
		File "/home/appveyor/DLTMP/tmpjm4hxin6", line 280, in <module>
		  main()
		File "/home/appveyor/DLTMP/tmpjm4hxin6", line 263, in main
		  json_out['return_val'] = hook(**hook_input['kwargs'])
		File "/home/appveyor/DLTMP/tmpjm4hxin6", line 114, in get_requires_for_build_wheel
		  return hook(config_settings)
		File "/home/appveyor/DLTMP/pip-build-env-pfgxta44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
		  return self._get_build_requires(config_settings, requirements=['wheel'])
		File "/home/appveyor/DLTMP/pip-build-env-pfgxta44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
		  self.run_setup()
		File "/home/appveyor/DLTMP/pip-build-env-pfgxta44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 480, in run_setup
		  super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
		File "/home/appveyor/DLTMP/pip-build-env-pfgxta44/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 311, in run_setup
		  exec(code, locals())
		File "<string>", line 15, in <module>
		File "/home/appveyor/projects/datalad-container/versioneer.py", line 1871, in get_version
		  return get_versions()["version"]
		File "/home/appveyor/projects/datalad-container/versioneer.py", line 1803, in get_versions
		  cfg = get_config_from_root(root)
		File "/home/appveyor/projects/datalad-container/versioneer.py", line 416, in get_config_from_root
		  parser.get("versioneer", "VCS")  # raise error if missing
		File "/usr/lib/python3.8/configparser.py", line 781, in get
		  d = self._unify_values(section, vars)
		File "/usr/lib/python3.8/configparser.py", line 1149, in _unify_values
		  raise NoSectionError(section) from None
	  configparser.NoSectionError: No section: 'versioneer'

</details>

I hope that with this explicit check it would be more informative crash.

Please neat-pick the diff since I have not tested it in action

### CODE REVIEW

File: src/header.py [hunk 1]
```diff
@@ -105,14 +105,26 @@ def get_config_from_root(root: str) -> VersioneerConfig:
...
+            if 'tool.versioneer' in pyproject_toml.read_text():
+                raise RuntimeError(
+                    f"Found tool.versioneer in {pyproject_toml} "
+                    "but the required dependency 'tomli' to read that file is "
+                    "not installed")
+            else:
+                print(
+                    "No section tool.versioneer found in {pyproject_toml} "
+                    "thus we will not try to load it",
+                    file=sys.stderr))
```
Comment (Callek) [inline src/header.py:119]: looks like CI is failing due to this extra close paren.

**Review by Callek — CHANGES_REQUESTED** [review]
I like the overall idea of testing for `tomli` and failing if not present. Though I am not certain I'm a huge fan of the wording of this, nor the raw text reading of pyproject.toml.

That said, I'm likely to accept this as-is if you can fix CI.

Alternatively I may take a shot at this fix myself in the next week or two based on your approach.

Thanks for the PR.

File: src/header.py (hunk 1 above)
Comment (yarikoptic) [reply src/header.py:119]: dang, I thought I fixed for that or was it elsewhere? ... ;) elsewhere... fixing....

**Review by Callek — APPROVED** [review]
Sorry for the delay, looks good

**Pull request merged**